addons/website_slides/models/slide_channel.py

```diff
@@ -533,26 +517,48 @@ def _compute_has_requested_access(self):
+    @api.depends('channel_partner_all_ids.partner_id', 'channel_partner_all_ids.member_status', 'channel_partner_all_ids.active')
     @api.depends_context('uid')
-    @api.model
     def _compute_membership_values(self):
         if self.env.user._is_public():
             self.is_member = False
             self.is_member_invited = False
             return
 
-        channel_partners = self.env['slide.channel.partner'].sudo().search([
-            ('channel_id', 'in', self.ids),
-            ('partner_id', '=', self.env.user.partner_id.id),
-        ])
-        active_channel_partners = channel_partners.filtered(
-            lambda channel_partner: channel_partner.member_status != 'invited')
-        invitation_pending_channels = (channel_partners - active_channel_partners).channel_id
-        active_channels = active_channel_partners.channel_id
+        data = {
+            member_status: channel_ids
+            for member_status, channel_ids in self.env['slide.channel.partner'].sudo()._read_group(
+                [('partner_id', '=', self.env.user.partner_id.id), ('channel_id', 'in', self.ids), ('active', '=', True)],
+                ['member_status'], ['channel_id:array_agg']
+            )
+        }
+        active_channels_ids = data.get('joined', []) + data.get('ongoing', []) + data.get('completed', [])
+        invitation_pending_channels_ids = data.get('invited', [])
         for channel in self:
-            channel.is_member = channel in active_channels
-            channel.is_member_invited = channel in invitation_pending_channels
+            channel.is_member = channel.id in active_channels_ids
+            channel.is_member_invited = channel.id in invitation_pending_channels_ids
+
+    def _search_is_member(self, operator, value):
+        if operator not in ['=', '!='] or not isinstance(value, bool):
+            raise NotImplementedError(_('Operation not supported'))
+        check_has_access = operator == '=' and value or operator == '!=' and not value
+
+        invited_courses_ids = self.env['slide.channel.partner'].sudo()._read_group(
+            [('partner_id', '=', self.env.user.partner_id.id), ('member_status', '!=', 'invited'), ('active', '=', True)],
+            aggregates=['channel_id:array_agg']
+        )
+        return [('id', 'in' if check_has_access else 'not in', invited_courses_ids[0][0])]
+
+    def _search_is_member_invited(self, operator, value):
+        if operator not in ['=', '!='] or not isinstance(value, bool):
+            raise NotImplementedError(_('Operation not supported'))
+        check_has_access = operator == '=' and value or operator == '!=' and not value
+
+        member_courses_ids = self.env['slide.channel.partner'].sudo()._read_group(
+            [('partner_id', '=', self.env.user.partner_id.id), ('member_status', '=', 'invited'), ('active', '=', True)],
+            aggregates=['channel_id:array_agg']
+        )
```

Comment:
Could maybe call a ``_search_is_member_invite_status``-like method, as code in both search is the same except a = / != ?